examples/nft-d/nft.go

```diff
@@ -0,0 +1,390 @@
+// parameters because token owner can be deduced from token ID itself.
+func TransferDivisible(from, to interop.Hash160, amount int, token []byte, data interface{}) bool {
+	if len(from) != 20 {
+		panic("invalid 'to' address")
+	}
+	if len(to) != 20 {
+		panic("invalid 'to' address")
+	}
+	if amount < 0 {
+		panic("negative 'amount'")
+	}
+	if amount > multiplier {
+		panic("invalid 'amount'")
+	}
+	ctx := storage.GetContext()
+	if !isValid(ctx, token) {
+		panic("unknown token")
+	}
+
+	// Note that although calling script hash is not checked explicitly in
+	// this contract it is in fact checked for in `CheckWitness` itself.
+	if !runtime.CheckWitness(from) {
+		return false
+	}
+
+	key := mkBalanceKey(from, token)
+	balance := getBalanceOf(ctx, key)
+	if amount > balance {
+		return false
+	}
+
+	if string(from) != string(to) {
+		updBalance := addToBalance(ctx, from, token, -amount)
+		if updBalance == 0 {
+			removeToken(ctx, from, token)
+			removeOwner(ctx, token, from)
+		}
+
+		updBalance = addToBalance(ctx, to, token, amount)
+		if updBalance != 0 {
+			addToken(ctx, to, token)
+			addOwner(ctx, token, to)
+		}
+	}
+	postTransfer(from, to, token, amount, data)
+	return true
+}
+
+// OwnerOf returns owner of specified token.
+func OwnerOf(token []byte) iterator.Iterator {
+	ctx := storage.GetReadOnlyContext()
+	if !isValid(ctx, token) {
+		panic("unknown token")
+	}
+	return ownersOf(ctx, token)
+}
+
+// BalanceOfDivisible returns the number of token with the specified tokenID owned by specified address.
+func BalanceOfDivisible(holder interop.Hash160, token []byte) int {
+	if len(holder) != 20 {
+		panic("bad owner address")
+	}
+	ctx := storage.GetReadOnlyContext()
+	key := mkBalanceKey(holder, token)
+	return getBalanceOf(ctx, key)
+}
+
+// end of divisible methods.
+
+// ownersOf returns iterator over owners of the specified token. Owner is
+// stored as value of the token key (prefix + token ID + owner).
+func ownersOf(ctx storage.Context, token []byte) iterator.Iterator {
+	key := mkTokenPrefix(token)
+	iter := storage.Find(ctx, key, storage.ValuesOnly)
+	return iter
+}
+
+func addOwner(ctx storage.Context, token []byte, holder interop.Hash160) {
+	key := mkTokenKey(token, holder)
+	storage.Put(ctx, key, holder)
+}
+
+func removeOwner(ctx storage.Context, token []byte, holder interop.Hash160) {
+	key := mkTokenKey(token, holder)
+	storage.Delete(ctx, key)
+}
+
+// OnNEP17Payment mints tokens if at least 10 GAS is provided. You don't call
+// this method directly, instead it's called by GAS contract when you transfer
+// GAS from your address to the address of this NFT contract.
+func OnNEP17Payment(from interop.Hash160, amount int, data interface{}) {
+	if string(runtime.GetCallingScriptHash()) != gas.Hash {
+		panic("only GAS is accepted")
+	}
+	if amount < 10_00000000 {
+		panic("minting SHARES costs at least 10 GAS")
+	}
+	tokenInfo := data.([]interface{})
```

Comment:
Why do you use array wrapper here?

Reply:
I had several fields before, but then only one had left. Now it's several fields again, so still aray.